src/lib/server/prisma/neovimconfigs/service.ts

```diff
@@ -2,585 +2,597 @@ import { Prisma, type NeovimConfig, type WhereType } from '@prisma/client';
-			user: {
-				select: {
-					avatarUrl: true
-				}
-			},
-			_count: {
-				select: {
-					neovimConfigPlugins: true
-				}
-			},
-			neovimConfigPlugins: {
-				select: {
-					paths: true
-				},
-				where: {
-					plugin: {
-						owner,
-						name
-					}
-				}
-			}
-		},
-		where: {
-			neovimConfigPlugins: {
-				some: {
-					plugin: {
-						owner,
-						name
-					}
-				}
-			}
-		},
-		orderBy: [
-			{
-				stars: 'desc'
-			},
-			{
-				repo: 'asc'
-			},
-			{
-				root: 'asc'
-			}
-		],
-		take
-	});
-	return configs.map(attachMetaData);
+  const configs = await prismaClient.neovimConfig.findMany({
+    include: {
+      user: {
+        select: {
+          avatarUrl: true
+        }
+      },
+      _count: {
+        select: {
+          neovimConfigPlugins: true
+        }
+      },
+      neovimConfigPlugins: {
+        select: {
+          paths: true
+        },
+        where: {
+          plugin: {
+            owner,
+            name
+          }
+        }
+      }
+    },
+    where: {
+      neovimConfigPlugins: {
+        some: {
+          plugin: {
+            owner,
+            name
+          }
+        }
+      }
+    },
+    orderBy: [
+      {
+        stars: 'desc'
+      },
+      {
+        repo: 'asc'
+      },
+      {
+        root: 'asc'
+      }
+    ],
+    take
+  });
+  return configs.map(attachMetaData);
 }
 
+
 export async function getConfigBySlug(
-	owner: string,
-	slug: string
+  owner: string,
+  slug: string
 ): Promise<NeovimConfigWithMetaData> {
-	const config = await prismaClient.neovimConfig.findFirstOrThrow({
-		include: {
-			syncs: {
-				orderBy: {
-					syncedAt: 'desc'
-				},
-				take: 1,
-				select: {
-					sha: true
-				}
-			},
-			user: {
-				select: {
-					avatarUrl: true
-				}
-			},
-			_count: {
-				select: {
-					neovimConfigPlugins: true
-				}
-			}
-		},
-		where: {
-			slug,
-			user: {
-				username: owner
-			}
-		},
-		orderBy: [
-			{
-				stars: 'desc'
-			},
-			{
-				repo: 'asc'
-			},
-			{
-				root: 'asc'
-			}
-		]
-	});
-	return attachMetaData(config);
+  const config = await prismaClient.neovimConfig.findFirstOrThrow({
+    include: {
+      syncs: {
+        orderBy: {
+          syncedAt: 'desc'
+        },
+        take: 1,
+        select: {
+          sha: true
+        }
+      },
+      user: {
+        select: {
+          avatarUrl: true
+        }
+      },
+      _count: {
+        select: {
+          neovimConfigPlugins: true
+        }
+      }
+    },
+    where: {
+      slug,
+      user: {
+        username: owner
+      }
+    },
+    orderBy: [
+      {
+        stars: 'desc'
+      },
+      {
+        repo: 'asc'
+      },
+      {
+        root: 'asc'
+      }
+    ]
+  });
+  return attachMetaData(config);
+}
+
+export async function deleteNeovimConfig(id: number, userId: number) {
+  return await prismaClient.neovimConfig.delete({
+    where: {
+      id_userId: {
+        id,
+        userId
+      }
```

Comment:
Ok makes sense, maybe this index is better than doing:

```
where: {
    id,
    userId,
}
```

Reply:
Yes. If I don't use an index prisma can't find the unique one by `id` and `userId`.